### PR TITLE
feat: Introduce `registerCollapsibleNavHeader` to ChromeService

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
This commit introduces an enhancement to the ChromeService class within our core application. It adds a new method named `registerCollapsibleNavHeader`, allowing plugins to customize the rendering of the collapsible navigation header in the global chrome UI.

With this new capability, plugins can now register their own rendering logic for the collapsible navigation header. This feature enhances the extensibility of our core system, empowering plugins to provide a more tailored and user-friendly navigation experience.

Key changes in this commit include:

1. The addition of the `registerCollapsibleNavHeader` method to the ChromeService class.
2. Appropriate updates to tests and typings to support the newly introduced functionality.

### Why introducing this change?
The upcoming workspace feature will bring noticeable change to the left navigation menu. One newly added component is the "Workspace Selector" component. This is a context menu on the top of the left navigation bar which enables easy navigation among workspaces. This PR tries to extend the chrome module with a generic method for plugins to customize the navigation bar's header.

I'm more seeking for suggestions/different opinions with this PR, please feel free to give feedback if you have different implementation ideas.

image

### Issues Resolved

Partially resolved: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5091

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ruanyl** [comment]
@ashwin-pc Thanks for the feedbacks! About this comment:
> The one drawback with your API though is that another plugin can override your registration.

You're right, I'm actually aware of this. There could potentially two options:
1. check the existence of the render, and throw error/warning if someone tries to override it
2. make the header to accept multiple customer renders

However, I'm not sure which one will best suite the real use case, to avoid over-optimize at the early stage, I keep it like this for now as currently there is only one use case from workspace plugin. We can always come back to this later.

**Comment by ashwin-pc** [comment]
Good to know that you thought of it :) Yeah i like your first option to add a check during registration to throw an error if registration is attempted twice. This way we can at least know when this happens.

**Review by zengyan-amazon — COMMENTED** [review]
LGTM

**Comment by ruanyl** [comment]
@ashwin-pc I added `console.warn` output when the header render was overridden, let me know if that looks good to you :)

**Review by ashwin-pc — DISMISSED** [review]
LGTM, @ruanyl can you resolve the conflicts in the changelog ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -941,4 +942,4 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### 🔩 Tests
 
-- Update caniuse to fix failed integration tests ([#2322](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2322))
+- Update caniuse to fix failed integration tests ([#2322](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2322))
```
Comment (SuZhou-Joe) [inline CHANGELOG.md:945]: Nit: unnecessary change on this line I suppose.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ashwin-pc — DISMISSED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]
